src/indicators/drought/chirpsgefs_rainfallforecast.py

```diff
@@ -0,0 +1,227 @@
+            else:
+                logger.debug(f"Retrieving stats for {date_str}")
+                rds = rioxarray.open_rasterio(chirpsgefs_raster_filepath)
+                df = compute_raster_stats(
+                    rds.sel(band=1), rds.rio.transform(), gdf_adm, adm_col
+                )
+                df.loc[:, "date"] = d
+                df.loc[:, "date_forec_end"] = df.loc[:, "date"] + timedelta(
+                    days=days_ahead - 1
+                )
+                df.drop("geometry", axis=1).to_csv(output_path, index=False)
+
+
+def get_rainy_season_dates(
+    rainy_season_path, onset_col="onset_date", cessation_col="cessation_date"
+):
+    """
+    Compute a datetimeindex with all dates across all rainy seasons.
+    This is used to only download data for these dates, since the files are large
+    Args:
+        rainy_season_path: filepath to csv with end and start dates of the rainy season, possibly per admin
+    """
+    df_rain = pd.read_csv(rainy_season_path, parse_dates=[onset_col, cessation_col])
+
+    # remove entries where there is no onset and no cessation date, i.e. no rainy season
+    df_rain = df_rain[
+        (df_rain.onset_date.notnull()) | (df_rain.cessation_date.notnull())
+    ]
+    # if onset date or cessation date is missing, set it to Nov 1/Jul1 to make sure all data of that year is downloaded. This happens if e.g. the rainy season hasn't ended yet
+    df_rain[df_rain.onset_date.isnull()] = df_rain[df_rain.onset_date.isnull()].assign(
+        onset_date=lambda df: pd.to_datetime(f"{df.season_approx.values[0]}-11-01")
+    )
+    df_rain[df_rain.cessation_date.isnull()] = df_rain[
+        df_rain.cessation_date.isnull()
+    ].assign(
+        cessation_date=lambda df: pd.to_datetime(
+            f"{df.season_approx.values[0]+1}-07-01"
+        )
+    )
+
+    # get min onset and max cessation for each season across all admins
+    df_rain_seas = df_rain.groupby("season_approx", as_index=False).agg(
+        {onset_col: np.min, cessation_col: np.max}
+    )
+
+    # create a daterange index including all dates within all rainy seasons
+    all_dates = pd.Index([])
+    for i in df_rain_seas.season_approx.unique():
+        seas_range = pd.date_range(
+            df_rain_seas[df_rain_seas.season_approx == i].onset_date.values[0],
+            df_rain_seas[df_rain_seas.season_approx == i].cessation_date.values[0],
+        )
+        all_dates = all_dates.union(seas_range)
+    return all_dates
+
+
+def download_chirpsgefs(date, config, days_ahead, use_cache=True):
+    """
+    Download the chirps-gefs africa forecast for the given year and day of the year
+    We are focussing on the Africa data, since global data gets massive. Nevertheless, even for Africa it gets massive.
+    Note that a part of the data for 2020 is missing due to a change in model
+    days: number of days forecast predicts ahead. When using 15 day, set to empty string
+    """
+
+    date_str = date.strftime("%Y%m%d")
```

Comment:
The dates in the filenames would be a bit more readable with hyphens (but don't worry about changing this now!)

Reply:
Also, the date format should be made a constant, since it's used in multiple places